static-instrumenter/agent-instrumenter/src/main/java/io/opentelemetry/contrib/staticinstrumenter/internals/ClassArchive.java

```diff
@@ -71,11 +71,21 @@ private InputStream getInputStreamForEntry(JarEntry inEntry, ZipEntry outEntry)
     ArchiveEntry entry = ArchiveEntry.fromZipEntryName(inEntry.getName());
     if (entry.shouldInstrument()) {
       String className = entry.getName();
-      byte[] modified = instrumentedClasses.get(entry.getPath());
-      if (modified != null) {
-        logger.debug("Found instrumented class: " + className);
-        entryIn = new ByteArrayInputStream(modified);
-        outEntry.setSize(modified.length);
+      try {
+        // TODO: load classes on archive's separate classloader
+        Class.forName(className, false, ClassLoader.getSystemClassLoader());
```

Comment:
Is this `Class.forName` call triggering the otel javaagent transformer? Can you add an explanation somewhere in the code, how does the transformation happen?

Reply:
Thanks!